executor/sort.go

```diff
@@ -479,7 +481,7 @@ func (e *TopNExec) processChildChk(childRowChk *chunk.Chunk) error {
 		next = childRowChk.GetRow(i)
 		if e.chkHeap.greaterRow(heapMax, next) {
 			// Evict heap max, keep the next row.
-			e.rowPtrs[0] = e.rowChunks.AppendRow(childRowChk.GetRow(i))
+			e.rowPtrs[0] = e.rowChunks.AppendRowByColIdxs(childRowChk.GetRow(i), e.columnIdxsUsedByChild)
```

Comment:
Should append all columns here. Otherwise, after columns pruned, columns used for sorting only will have no values, and lead to wrong result.

Reply:
@pingyu PTAL, The following ideas is my understanding of review. Is that so?

1. prune unnecessary columns in planner (pass a thinner schema)
2. revert columns pruning on chunk process

Reply:
1. Yes.
2. What does this mean?

Reply:
@pingyu `Append all columns here`